### DISCUSSION TITLE
Manage Class Feature witht several effect depending on the level

### DISCUSSION THREAD

**Discussion by dcpc007** [Rughalt/D35E#660] [Show and tell]
For a custom class, i have a feature the give several bonuses, growing with levels, but not the same effect.

Like : at level 1, gain +2 STR, at level 3, gain a special attack which xxxxx

I try to mind how to render that. 
I choose this form : 
- I create 2 Class feature : 
  - MyFeature (1) : add +2 STR
  - MyFeature (3) : add special attack
then in the class i can add each effect at the indicated level of the class.

If you think there's other (better) way, feel free to comment.

-----
Another example : i have a boost on a creature that give +4 by default, and later 2 extensions that push +2 more the bonus.
I create 3 effects and associated Class features  : 
- Feature : (at level 1) buff +4
- Improved Feature : (at level 4), buff +6 (with the portion of text mnetionning it's a cumulative +2 of the precedent buff)
- Superior Feature) : (at level 7), buff +8 (idem)

Then i have 2 solution to apply it :
- Manually apply the correct buff when needed on a target (+4, +6 or +8)
- If frequently used, like for a personnal summoned monster or a companion, add the 3 buffs in their sheet, with only the current level of boost = Active. And .. when my character gain level, i'll have to edit each target prototype sheets to put the new buff active by default.

**Comment by Rughalt** [comment]
The first one is how this should be done. These are separate features and should have separate entries.

The second one can be done using ternary operators inside specific changes, like `condition ? value if true : value if false` For example `@attributes.hd.total > 10 ? 2 : 1` will result in 2 if `@attributes.hd.total` is greater than 10 and 1 if not.
